SumoLogic.Logging.AspNetCore/LoggerFactoryExtensions.cs

```diff
@@ -0,0 +1,48 @@
+﻿using System;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace SumoLogic.Logging.AspNetCore
+{
+    public static class LoggerFactoryExtensions
```

Comment:
Regarding this extensions - I think another approach (for registration) would be better, and more correct. Instead of calling Services.AddSingleton<ILoggerProvider, LoggerProvider>(), we can add our provider to .net core ILoggerFactory.
For example,
public static ILoggerFactory AddSumoLogic(this ILoggerFactory factory, string uri)
        {
            factory.AddProvider(SumoLoggerProvider.Create(uri)); // SumoLoggerProvider is //SumoLogic.Logging.AspNetCore.LoggerProvider
        }

And then use it like this:
public void Configure(IApplicationBuilder app, IHostingEnvironment env, ILoggerFactory loggerFactory)
{
    // Log to SumoLogic.
    loggerFactory.AddSumoLogic("my-SumoLogic-uri");
...
}
In such way, we will not break any other registered providers, and will not substitute ILoggerProvider registration in build .net core DI.
What do you think ?

Reply:
TBH, this part of code is coming from examples in [this post](https://andrewlock.net/creating-a-rolling-file-logging-provider-for-asp-net-core-2-0/) and looks like the author think:

> In ASP.NET Core 2.0, logging providers are added using DI, so adding our new logging provider just requires adding the FileLoggerProvider to DI, as in the AddFile() method above.

which is a benefit? thought?

Reply:
I will reference implementation for log4net (I hope you don't mind :))
https://github.com/huorswords/Microsoft.Extensions.Logging.Log4Net.AspNetCore/tree/develop/src/Microsoft.Extensions.Logging.Log4Net.AspNetCore
With their approach, each time you are requesting ILogger<T>instance through constructor: 
1. ILoggerFactory -> ILogger CreateLogger(string categoryName); will be called - to create ILogger 
2. This ILogger will contains all the Loggers, from registered LoggingProvider(s).
With your approach, each time you are requesting ILogger<T>instance through constructor: 
1. ILoggerFactory -> ILogger CreateLogger(string categoryName); will be called - to create ILogger 
2. This ILogger will not contain SumoLogicLogger, because SumoLogicLogginProvider was not registered inside ILoggerFactory.
builder.Services.AddSingleton<ILoggerProvider, LoggerProvider>(); - this will add LoggerProvider to DI, but it would not be resolved during ILogger<T> contraction.
Also, here is the reference to Microsoft.Extensions.Logging.LoggerFactory:
https://github.com/aspnet/Logging/blob/master/src/Microsoft.Extensions.Logging/LoggerFactory.cs

Reply:
added. so now we can using both approach as in Log4Net